### ISSUE TITLE
Config cache refresh issue with .env.local

### ISSUE THREAD

**Issue by plamenzhelyazkov** [laravel/framework#49896]
### Laravel Version

10.33.0

### PHP Version

8.1.2

### Description

I'm using .env.local.

When rebuilding the config cache (artisan config:cache) it didn't use the .env.local if there is already created cache (bootstrap/cache/config.php) but instead it uses the parameters from the .env.

If I clear manually the cache (artisan config:clear) and rebuild it only once (artisan config:cache) it takes the parameters from the .env.local.

### Steps To Reproduce

1. Set variable in the .env - for example **MAIL_HOST=gmail.com**
2. Set the same variable in the .env.local - **MAIL_HOST=localhost**
3. Run `artisan config:clear`
4. Run `artisan config:cache`
5. See the content in the cache file (bootstrap/cache/config.php) - it is using the value from the .env.local
6. Run `artisan config:cache`
7. See the content in the cache file (bootstrap/cache/config.php) - **it is using the value from the .env**

**Comment by driesvints** [comment]
Hey there,

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!
